### PR TITLE
Made Miscellaneous Refactors

### PR DESCRIPTION
**Pull request by johnludwigm**
* Fixed f-strings missing placeholders. 
* Used comprehensions over for-loops
* Made some test infra more Pythonic
* Type-hinted a few things
* Fixed a variable that hadn't been defined
* Removed unneeded imports
* `sorted` automatically converts the iterable to a list, so `sorted(list(set(x)))` is the same as `sorted(set(x))`

Thanks `flake8`!

Currently updating my computer at work. Figured I'd submit this.

### CODE REVIEW

File: cl/search/tasks.py [hunk 1]
```diff
@@ -548,7 +547,7 @@ def get_doc_from_es(
             )
         except (ConflictError, RequestError) as exc:
             logger.error(
-                f"Error indexing the {es_document.Django.model.__name__.capitalize()} with ID: {main_instance_id}. "
```
Comment (johnludwigm) [inline cl/search/tasks.py:551]: This variable was never defined.

File: cl/search/tasks.py [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import logging
 import socket
-from collections import deque
```
Comment (johnludwigm) [inline cl/search/tasks.py:3]: Unused

**Review by mlissner — APPROVED** [review]
Cool, sure, looks good to me, thanks.

**Comment by mlissner** [comment]
A couple small comments:

1. We have a standard for commit messages. I don't usually care too much about it, but since this is a toe-in-the-water PR I might as well mention that it exists. See item 3, here: https://github.com/freelawproject/courtlistener/wiki/Getting-Started-Developing-CourtListener/0c803bbbb92eac2fac2f5d79dbff4689f001acc2#guidelines-for-contributions

    No need to clean up old commits though. If you do that, I'll have to re-review your code after your force push, which I'd rather not do.

2. We need you to sign our CLA (see the message from the bot above).

Once you've got the CLA signed, I'll enable auto-merge and we'll be off to the races. Thanks for your first PR!

**Comment by johnludwigm** [comment]
Do I have to do something to get it merged?

**Comment by mlissner** [comment]
> Do I have to do something to get it merged?

My fault. I thought it'd tell me when you signed the CLA. You're good now and we're merged, thank you!

**Pull request merged**